db/post_migrate/20201111005205_move_data_to_event_dates.rb

```diff
@@ -51,9 +51,14 @@ def due_reminders(event)
     return [] if event.reminders.blank?
     event.reminders.split(",").map do |reminder|
       value, unit = reminder.split('.')
-      date = event.original_starts_at - value.to_i.send(unit)
-      { description: reminder, date: date }
-    end.select { |reminder| reminder[:date] <= Time.current }.sort_by { |reminder| reminder[:date] }
+
+      begin
+        date = event.original_starts_at - value.to_i.public_send(unit)
+        { description: reminder, date: date }
+      rescue NoMethodError
+        nil
+      end
```

Comment:
This is still vulnerable to the security issue, right? How about

```suggestion
      allowed = ["years", "months", "weeks", "days", "hours", "minutes", "seconds"]
      next if allowed.include?(unit)
      date = event.original_starts_at - value.to_i.public_send(unit)
      { description: reminder, date: date }
```
(`allowed` could also be moved into a constant at the top of the file)

Reply:
it seems like it's very improbable it would happen or do any arm with the public send and the fact that should be with vulnerable data on something that's hasn't been migrated yet... but yes I can also add extra safety here

Reply:
Agreed it's improbable, but it's not impossible ;)